### PR TITLE
Adds pruno, aka prison wine, to the game.

### PR DESCRIPTION
**Pull request by spessbandit**
## About The Pull Request

Prisoners -- or anybody, but traditionally it is done by prisoners -- have the ability to brew prison wine, also known as pruno, also known as [this delightful stuff](http://www.blacktable.com/gillin030901.htm).

pruno

The recipe currently is: 4 fruits/grown produce of any kind, 2 candy corn, 1 piece of moldy bread, 15 water, and 1 garbage bag. This makes pruno mix. Put the pruno mix into a structure or machine, (read: the toilet, the washing machine) and after a set amount of time -- right now it's at 30 seconds  -- it becomes the final product.

Pruno, the drink, is extremely alcoholic and extremely gross, and also has a small chance of being toxic as well. It does give you the positive alcohol moodlet though. (I may change this, I want the net mood effect to be slightly positive.)

Because pruno is supposed to be prison content, i.e., craftable in perma, using only things located in perma, this also makes a number of minor changes to the game/maps to do this:
- Moldy bread is a new item. It has a recipe in case someone wants to make it for some reason, but it is also sold in the sustenance vendor in perma.
- Perma now has a garbage bag. This should not affect balance because perma already has a number of places to hide items.
- All toilet cisterns in perma are open, because prisoners otherwise have no way of crowbarring them open. See above note about balance.

## Why It's Good For The Game

It gives prisoners something else to do, and possibly gives the Warden something else to do, because traditionally this is contraband that should be confiscated.

## Changelog
:cl: bandit
add: The uptick of long-term detainees in Nanotrasen brigs has resulted in prisoners reviving the ancient, illegal tradition of brewing prison wine from fruit, mold, water, and despair. Security personnel should keep an eye out for this contraband.
tweak: All Nanotrasen permanent confinement facilities have been given garbage bags, and prisoners have been allotted bread with their meals. This has absolutely nothing to do with the above.
/:cl:

### CODE REVIEW

**Comment by Bawhoppen** [comment]
Loving it. May I implore you to just make the cisterns openable by hand though? Shouldn't be much extra work...

**Comment by Farquaar** [comment]
Next Week: Jankum added to /tg/station

**Comment by spessbandit** [comment]
if I remember correctly* there's some arcane reason why the cisterns can't be opened by hand

* I may not remember correctly

**Comment by Bawhoppen** [comment]
I always kind of wondered this myself, but after reviewing the code and history, I assume the reason is just because when it's open, touching it will retrieve the item from out of the cistern. 
Perhaps we could add it as an alt-click option to lift it on and off? Or maybe a storage slot inside the toilet like a wallsafe?

**Comment by HomocidlePancakes** [comment]
mmm... Toilet wine

**Comment by Krysonism** [comment]
Is possible to rename the pruno mix to something more generic like "fermentation slurry", or "mouldy slurry" to give the reagent a bit more reusability? 

I am thinking of using it as vat growing reagent.

**Comment by spessbandit** [comment]
once I go that route I have to consider what happens with the existing fermentation barrel (if you're wondering why I didn't work off that, that takes individual fruits, not recipes, and also where the fuck do you put build in perma), and the recipe is really pretty specific

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm [hunk 1]
```diff
@@ -2205,3 +2205,18 @@ All effects don't start immediately, but rather get worse over time; the rate is
...
+	color = "#E78108"
+	description = "Fermented prison wine made from fruit, sugar, and despair. Security loves to confiscate this, which is the only kind thing Security has ever done."
+	boozepwr = 85
+	taste_description = "your tastebuds being individually shanked"
+	glass_icon_state = "glass_orange"
+	glass_name = "glass of pruno"
+	glass_desc = "Fermented prison wine made from fruit, sugar, and despair. Security loves to confiscate this, which is the only kind thing Security has ever done."
+
+/datum/reagent/consumable/ethanol/pruno/on_mob_life(mob/living/carbon/M)
+	SEND_SIGNAL(M, COMSIG_ADD_MOOD_EVENT, "gross_food", /datum/mood_event/gross_food)
```
Comment (Qustinnus) [inline code/modules/reagents/chemistry/reagents/alcohol_reagents.dm]: Use disgust

**Comment by Krysonism** [comment]
> once I go that route I have to consider what happens with the existing fermentation barrel (if you're wondering why I didn't work off that, that takes individual fruits, not recipes, and also where the fuck do you put build in perma), and the recipe is really pretty specific

For clarification I am not suggesting you to change anything about the mechanics, I like this PR  I just think having a less specific name for the precursor reagent would improve reusability.

Let's say a coder wants to add blue cheese in the future, if there is a reagent called "mouldy slurry", great, they already have a fitting reagent they can use. It is harder to justify pruno mix having other applications than.. making pruno.

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm (hunk 1 above)
Comment (spessbandit) [reply code/modules/reagents/chemistry/reagents/alcohol_reagents.dm]: the thing is, it's intended to be an actual toxin (in real life botulism, we don't really have that)

**Comment by spessbandit** [comment]
I mean what I should do is just implement mold but that's a separate pr

**Comment by spessbandit** [comment]
or to clarify:

pruno mix : fruit + sugar and source of yeast/mold
what you are suggesting: just source of yeast/mold

**Comment by evilcyborgmrt** [comment]
This shouldn't require candy corn

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm (hunk 1 above)
Comment (Qustinnus) [reply code/modules/reagents/chemistry/reagents/alcohol_reagents.dm]: then why does it just send a moodlet

**Comment by Qustinnus** [comment]
@spessbandit the name is too specific though it implies it can only be used for pruno. just name it (something)slurry or whatever

**Comment by spessbandit** [comment]
what the fuck else would you make out of rotten fruit and candy, certainly not blue cheese

**Comment by spessbandit** [comment]
@evilcyborg there is no sugar in perma, so the closest thing is candy, available in the sustenance vendor.

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm (hunk 1 above)
Comment (spessbandit) [reply code/modules/reagents/chemistry/reagents/alcohol_reagents.dm]: oh wait shit you’re right, was mixing this up with the badfood toxin

**Comment by spessbandit** [comment]
new changes:
- makes the moodlet disgust, not bad food
- breaks mold out into a separate reagent, per above; it currently does not have a source other than moldy bread, but that's for a different PR
- adds a name var to /grown base type so they don't show up in the recipe as "Snack"

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm [hunk 2]
```diff
@@ -2205,3 +2205,18 @@ All effects don't start immediately, but rather get worse over time; the rate is
...
+	color = "#E78108"
+	description = "Fermented prison wine made from fruit, sugar, and despair. Security loves to confiscate this, which is the only kind thing Security has ever done."
+	boozepwr = 85
+	taste_description = "your tastebuds being individually shanked"
+	glass_icon_state = "glass_orange"
+	glass_name = "glass of pruno"
+	glass_desc = "Fermented prison wine made from fruit, sugar, and despair. Security loves to confiscate this, which is the only kind thing Security has ever done."
+
+/datum/reagent/consumable/ethanol/pruno/on_mob_life(mob/living/carbon/M)
+	SEND_SIGNAL(M, COMSIG_ADD_MOOD_EVENT, "disgust", /datum/mood_event/verygross)
```
Comment (Qustinnus) [inline code/modules/reagents/chemistry/reagents/alcohol_reagents.dm]: dont do this, add disgust to the mob

Comment (spessbandit) [reply code/modules/reagents/chemistry/reagents/alcohol_reagents.dm]: isn't that what I'm doing

Comment (spessbandit) [reply code/modules/reagents/chemistry/reagents/alcohol_reagents.dm]: (this isn't meant to be rude, it's a genuine question)

Comment (Qustinnus) [reply code/modules/reagents/chemistry/reagents/alcohol_reagents.dm]: there's a seperate system for disgust similar to how things like drugginess work. that automatically assigns the moods etc and makes you puke after a while

Comment (trollbreeder) [reply code/modules/reagents/chemistry/reagents/alcohol_reagents.dm]: use adjust_disgust() and an if prob to randomly give spikes of DISGUSTANG
```suggestion
	if(prob(30)) // dont wanna spam disgust
		M.adjust_disgust(15)
```

Comment (Qustinnus) [reply code/modules/reagents/chemistry/reagents/alcohol_reagents.dm]: the prob doesn't make sense. just lower the amount from 15 to 5 and you have the same effect without the RNG

File: code/modules/food_and_drinks/food/condiment.dm [hunk 3]
```diff
@@ -227,8 +227,6 @@
 	icon_state = "mayonnaise"
 	list_reagents = list(/datum/reagent/consumable/mayonnaise = 50)
 
-
```
Comment (spookydonut) [inline code/modules/food_and_drinks/food/condiment.dm]: avoid doing this in future

File: code/modules/food_and_drinks/drinks/drinks/bottle.dm [hunk 4]
```diff
@@ -537,3 +537,44 @@
...
+/obj/item/reagent_containers/food/drinks/bottle/pruno/Initialize()
+	. = ..()
+	addtimer(CALLBACK(src, .proc/check_fermentation), fermentation_time)
+
+// Checks to see if the pruno can ferment, i.e. is it inside a structure (e.g. toilet), or a machine (e.g. washer)?
+// if so, start countdown to ferment, if not, check again in fermentation_time seconds
+// TODO: make it so the washer spills reagents if a reagent container is in there, for now, you can wash pruno
+
+/obj/item/reagent_containers/food/drinks/bottle/pruno/proc/check_fermentation()
+	if (!(istype(loc, /obj/machinery) || istype(loc, /obj/structure)))
```
Comment (spookydonut) [inline code/modules/food_and_drinks/drinks/drinks/bottle.dm]: you should be using movement signals to start/cancel fermentation, not checking every x seconds

**Review by spookydonut — CHANGES_REQUESTED** [review]

**Comment by spookydonut** [comment]
@spessbandit hmu on discord and i'll help you change this to use signals

**Comment by spookydonut** [comment]
nvm did it for you, merge this on your repo https://github.com/spessbandit/tgstation/pull/1

**Comment by spessbandit** [comment]
@spookydonut fuck just saw this have been in midterms hell will do

**Comment by spessbandit** [comment]
@spookydonut done

**Pull request merged**